cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -214,8 +222,12 @@ func upgradeComponent(component string, certsRenewMgr *renewal.Manager, waiter a
 		return err
 	}
 	if equal {
-		fmt.Printf("[upgrade/staticpods] Current and new manifests of %s are equal, skipping upgrade\n", component)
-		return nil
+		if !apiServerUpgrade {
+			fmt.Printf("[upgrade/staticpods] Current and new manifests of %s are equal, skipping upgrade\n", component)
+			return nil
+		}
+		fmt.Printf("[upgrade/staticpods] Current and new manifests of %s are equal, but %s will be upgraded\n",
```

Comment:
This may be confusing for users.
 If it's for debug purposes, it might be better to use `klog.V(4)` ?